src/vulkan/engine_vulkan.cc

```diff
@@ -72,17 +73,197 @@ Result EngineVulkan::InitDeviceAndCreateCommand() {
   return {};
 }
 
-Result EngineVulkan::Initialize(const std::vector<Feature>&,
+VkPhysicalDeviceFeatures EngineVulkan::RequiredFeatures(
```

Comment:
I think this should go away and we should just pass the std::vector<Features> into AreAllRequiredFeaturesAvailable. There is no reason to do this conversion.

Reply:
Updated.